presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -605,7 +609,24 @@ public static String computeBucketedFileName(String queryId, int bucket)
     public static String getFileExtension(JobConf conf, StorageFormat storageFormat)
     {
         // text format files must have the correct extension when compressed
-        if (!HiveConf.getBoolVar(conf, COMPRESSRESULT) || !HiveIgnoreKeyTextOutputFormat.class.getName().equals(storageFormat.getOutputFormat())) {
+        String outputStorageFormat = storageFormat.getOutputFormat();
```

Comment:
Can we pass it the file extension to `StorageFormat` and fetch from it ? On doing so we can remove the if statements..
In the `HiveStorageFormat` we can have a `Function<JobConf>` which generates the file extension based on JobConf (for text file format.) We can pass the same Function to `StorageFormat` and generate the required extension based on file type and its compression logic

Reply:
It would be better if we could have some tests for it